File: adventofcode/2015/day/02/aoc_15_day_2_gifts.cpp
#include <bits/stdc++.h>

int main()
{
  int l, w, h;
  unsigned long long a{0};
  while(scanf("%dx%dx%d", &l, &w, &h) != EOF)
  {
    //printf("l=%d w=%d h=%d\n", l, w, h);
    a += 2*l*w + 2*w*h + 2*h*l;
    a += std::min(std::min(l*w, l*h), w*h);
    //printf(">> a=%lu\n", a);
  }

  printf("Part one: %lu.\n", a);
}

File: adventofcode/2015/day/02/aoc_15_day_2_gifts.nim
import std/strscans

var l, w, h: int
var a = 0
while true:
  var line: string
  try:
    line = readLine(stdin)
  except:
    break
  if scanf(line, "$ix$ix$i", l, w, h):
    a += 2*l*w+2*l*h+2*w*h + min(l*w, min(l*h, w*h))
  else:
    break

echo "Part one: ", a
